### ISSUE TITLE
Cant pull Tools from Techniques

### ISSUE THREAD

**Issue by TiiTcHY** [swimlane/pyattck#51]
I am trying to extract techniques, sub techniques, tactics, actors and tools/software:

for technique in attack.enterprise.techniques:
    print(technique.id)
    print(technique.name)
    # you can also access generated data sets on aa technique
    print(technique.command_list)
    print(technique.commands)
    print(technique.queries)
    print(technique.datasets)
    print(technique.possible_detections)
    for subtechnique in technique.subtechniques:
        print(subtechnique.id)
        print(subtechnique.name)
    # accessing tactics that this technique belongs to
        for tactic in technique.tactics:
            print(tactic.id)
            print(tactic.name)

        # accessing mitigation recommendations for this technique
            for mitigation in technique.mitigations:
                print(mitigation.id)
                print(mitigation.name)

        # accessing actors using this technique
                for actor in technique.actors:
                    print(actor.id)
                    print(actor.name)
                    for tool in technique.tools:
                        print(tool.name)

However, i keep getting the following error AttributeError: 'AttckTechnique' object has no attribute 'tools'

Is there a way i can pull the tools as well as the following information. I am trying to export this into a CSV/JSON file to be able to read it. Is there a way to export this out as at the moment i cant get the fields to match up

**Comment by joshswimlane** [comment]
Hello @TiiTcHY so thanks for pointing this out and I found a bug that I have fixed and will push a new version out soon.  This is going to be part of the next major release which is 3.0.0.  

The pull request is this one:

#49 

Once that is distributable then you should be able to generate the desired CSV using this code.

```
from pyattck import Attck

attack = Attck()

csv_columns = ['technique_id', 'technique_name', 'command_list', 'commands', 'queries', 'datasets', 'possible_detections', 'subtechnique_id', 'subtechnique_name']
csv_row_list = []

for technique in attack.enterprise.techniques:
    csv_dict = {
        'technique_id': technique.id,
        'technique_name': technique.name,
        'command_list': technique.command_list,
        'commands': technique.commands, 
        'queries': technique.queries,
        'datasets': technique.datasets,
        'possible_detections': technique.possible_detections
    }
    count = 1
    for subtechnique in technique.subtechniques:
        id_key_name = 'subtechnique_id_{}'.format(count)
        name_key_name = 'subtechnique_name_{}'.format(count)
        csv_columns.append(id_key_name)
        csv_columns.append(name_key_name)
        csv_dict.update({
            id_key_name: subtechnique.id,
            name_key_name: subtechnique.name
        })
        count += 1
    count = 1
    # accessing tactics that this technique belongs to
    for tactic in technique.tactics:
        id_key_name = 'tactic_id_{}'.format(count)
        name_key_name = 'tactic_name_{}'.format(count)
        csv_columns.append(id_key_name)
        csv_columns.append(name_key_name)
        csv_dict.update({
            id_key_name: tactic.id,
            name_key_name: tactic.name
        })
        count += 1
    count = 1
    # accessing mitigation recommendations for this technique
    for mitigation in technique.mitigations:
        id_key_name = 'mitigation_id_{}'.format(count)
        name_key_name = 'mitigation_name_{}'.format(count)
        csv_columns.append(id_key_name)
        csv_columns.append(name_key_name)
        csv_dict.update({
            id_key_name: mitigation.id,
            name_key_name: mitigation.name
        })
        count += 1
    count = 1
    # accessing actors using this technique
    for actor in technique.actors:
        id_key_name = 'actor_id_{}'.format(count)
        name_key_name = 'actor_name_{}'.format(count)
        csv_columns.append(id_key_name)
        csv_columns.append(name_key_name)
        csv_dict.update({
            id_key_name: actor.id,
            name_key_name: actor.name
        })
        count += 1
        tool_count = 1
        for tool in actor.tools:
            name_key_name = 'tool_name_{}'.format(count)
            csv_columns.append(name_key_name)
            csv_dict.update({
                name_key_name: tool.name
            })
            tool_count += 1
    csv_row_list.append(csv_dict)

import csv

csv_file = "pyattck.csv"

try:
    with open(csv_file, 'w') as csvfile:
        writer = csv.DictWriter(csvfile, fieldnames=csv_columns)
        writer.writeheader()
        for data in csv_row_list:
            writer.writerow(data)
except IOError:
    print("I/O error")
```

Sorry you ran into issues and I'm actively working on resolving this now.

**Comment by TiiTcHY** [comment]
Amazing, thank you for sorting this out. Do you know when 3.0 will be released so I can pull this data? Thanks again

Kind Regards,

Daniel Whitcombe

________________________________

**Comment by TiiTcHY** [comment]
I have been trying the code that you kindly supplied to extract the data that i need (Thanks again) However, it appears to pull duplicates for the subtechnique multiple times?
![Capture](https://user-images.githubusercontent.com/43418761/106730919-cd971680-6606-11eb-8db5-ba0b71f9bbb0.PNG)

**Comment by MSAdministrator** [comment]
@TiiTcHY this has been fixed in pyattck 3.0.0 #49
